gnd/src/main/java/com/google/android/gnd/ui/common/AbstractDialogFragment.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gnd.ui.common
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import com.google.android.gnd.util.Debug
+import javax.inject.Inject
+
+abstract class AbstractDialogFragment : AppCompatDialogFragment() {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    /**
+     * Uses [ViewModelFactory] to obtain an instance of the view model of the specified class.
+     */
+    protected fun <T : ViewModel?> getViewModel(modelClass: Class<T>): T {
+        return viewModelFactory.get(this, modelClass)
```

Comment:
nit: could be expressed as a single-line expression function

Reply:
Done